### PR TITLE
bug: report error from image library

### PR DESCRIPTION
**Pull request by jrconlin**
Closes: #348

### CODE REVIEW

**Comment by ncloudioj** [comment]
@jrconlin When encountering this kind of error, is it possible for Contile to bail out rather than return an error code in the response? Doing so at least allows us to return some good tiles instead of nothing?

**Comment by jrconlin** [comment]
This is currently failing tests because of the test suite change for mobile endpoints. #346 
Rather than spend time correcting for this change, I'm going to wait until after the multi-endpoint change lands, then fix things here.

File: src/server/img_storage.rs [hunk 1]
```diff
@@ -190,9 +190,13 @@ impl StoreImage {
     pub fn meta(&self, image: &Bytes, fmt: ImageFormat) -> HandlerResult<ImageMetrics> {
         let mut reader = ImageReader::new(Cursor::new(image));
         reader.set_format(fmt);
-        let img = reader
-            .decode()
-            .map_err(|_| HandlerErrorKind::BadImage("Image unreadable"))?;
+        let img = reader.decode().map_err(|e| {
+            let mut tags = Tags::default();
+            tags.add_extra("error", &e.to_string());
```
Comment (pjenvey) [inline src/server/img_storage.rs:195]: I think both `url` and `format` are really worth adding as tags here too -- the easiest way to do that is to probably shuffle this method around a bit to have the same signature as `validate`.

**Review by pjenvey — CHANGES_REQUESTED** [review]

File: src/server/img_storage.rs (hunk 1 above)
Comment (jrconlin) [reply src/server/img_storage.rs:195]: Sigh. 
the change was included in the merge https://github.com/mozilla-services/contile/pull/349/commits/e767a61d02cb42123647db4604497a8e68c68ca3

File: src/adm/tiles.rs [hunk 2]
```diff
@@ -212,6 +212,7 @@ pub async fn get_tiles(
                     // We still want to track this as a server error later.
                     //
                     // TODO: Remove this after the shared cache is implemented.
+                    dbg!(&e);
```
Comment (pjenvey) [inline src/adm/tiles.rs:215]: forgot to kill this

**Review by pjenvey — CHANGES_REQUESTED** [review]

File: src/adm/tiles.rs (hunk 2 above)
Comment (jrconlin) [reply src/adm/tiles.rs:215]: argh... hang on.

**Review by pjenvey — APPROVED** [review]

**Pull request merged**